tests/nopanic.ci

```diff
@@ -6,16 +6,17 @@ Copyright 2021 Keylime Authors
 To prevent CI failing for approved instance of banned words, add a comment: //#[allow_ci]
 '''
 
-import os
+import subprocess
 
 banned = ["unwrap(", "panic!("]
 
-srcs = os.listdir("src")
+res = subprocess.run(["git", "ls-files", "--cached", "--exclude", "*.rs", "--ignored"], capture_output=True)
```

Comment:
IIUC this is equivalent to `find src/ -name '*.rs'`. I would like to not depend on `git` cli to run tests. To express this in Python it is as:

```python
$ python3
Python 3.10.8 (main, Oct 19 2022, 07:46:20) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> list(pathlib.Path("src/").glob("**/*.rs"))
[PosixPath('src/algorithms.rs'), PosixPath('src/ima_entry.rs'), PosixPath('src/notifications_handler.rs'), PosixPath('src/crypto.rs'), PosixPath('src/main.rs'), PosixPath('src/version_handler.rs'), PosixPath('src/serialization.rs'), PosixPath('src/keys_handler.rs'), PosixPath('src/config.rs'), PosixPath('src/revocation.rs'), PosixPath('src/error.rs'), PosixPath('src/common.rs'), PosixPath('src/ima.rs'), PosixPath('src/tpm.rs'), PosixPath('src/registrar_agent.rs'), PosixPath('src/secure_mount.rs'), PosixPath('src/errors_handler.rs'), PosixPath('src/quotes_handler.rs'), PosixPath('src/ima_emulator.rs'), PosixPath('src/permissions.rs')]
```

Reply:
A benefit of using `git` is that it allows us to ignore files not tracked in git, so you could run the check in pre-commit phase even if extra files are generated in the build process. I can switch to pathlib for this particular case.